lib/Internal/Windows/SocketConnector.php

```diff
@@ -353,7 +365,19 @@ public function onProcessConnectTimeout($watcher, Handle $handle) {
     }
 
     public function registerPendingProcess(Handle $handle) {
-        $handle->connectTimeoutWatcher = Loop::delay(self::CONNECT_TIMEOUT, [$this, 'onProcessConnectTimeout'], $handle);
+        // See onProcessConnectTimeout(), use double defer(), because defer() is always executed before timers.
+        $loopTicked = &$handle->loopTicked;
+        Loop::defer(static function () use (&$loopTicked) {
```

Comment:
What effect does `static` have here?

Reply:
It prevents `$this` being captured and thus the event loop having a reference to the object.